Presence of a directory named `example` on project top level causes the `flutter drive ..` integration test to fail.

# Issue description (tl;dr): 
Presence of a directory named `example` on project top level causes the `flutter drive ..` integration test to fail.
The integration test which fails 
```shell
  flutter drive --driver=test_driver/integration_test.dart --target=integration_test/screenshot_test.dart
```
is the screenshot test created exactly by following instructions in 
  https://github.com/flutter/flutter/tree/master/packages/integration_test#integration_test. 
The full description of how the project was created and how the test was run is below.
The failure is very specific to the directory name `example`. The contents of the directory does not matter - just 
the presence of such named directory. Directories with other names do not cause this issue.
**This failure prevents library projects to contain a example app, which by convention should be in a directory 
named `example`.**
 
I created a project on Github  https://github.com/mzimmerm/screenshot 
which can be used to fully duplicate the issue.
The project is a rudimentary Flutter example app created from command line,
with added files `test_driver/integration_test.dart` and `integration_test/screenshot_test.dart`
described in https://github.com/flutter/flutter/tree/master/packages/integration_test#integration_test. 
 
# To duplicate the issue

Install the duplication project from Github as follows:

On your local system, from the command line, run the following commands:
    
```shell
   cd ~/tmp # create if not exist
   git clone https://github.com/mzimmerm/screenshot.git
   cd screenshot
   # Note that the project top level does NOT contain a directory named `example`.
   
   # Clean and run the test (which creates a screenshot) from command line
   flutter clean
   flutter emulator --launch "Nexus_6_API_29_2" # Change the emulator name to one on your system 
   sleep 25 # make sure the emulator starts fully
   
   # Run the added Flutter integration (drive) test which creates a screenshot of the app
   flutter drive --driver=test_driver/integration_test.dart --target=integration_test/screenshot_test.dart
   # Expected: screenshot-1.png is created in the project top level
   # Actual:   screenshot-1.png is created in the project top level
   
   # Remove the screenshot to ensure we continue from the same initial state
   rm screenshot-1.png
   # Exit the emulator to ensure we continue from the same initial state.
   
   # Now create an empty directory `example` which will cause the same test to fail
   
   mkdir example
  
   # Run the same test as above again
   # Clean and run the test (which creates a screenshot) from command line
   flutter clean
   flutter emulator --launch "Nexus_6_API_29_2" # Change the emulator name to one on your system 
   sleep 25
   flutter drive --driver=test_driver/integration_test.dart --target=integration_test/screenshot_test.dart
   # Expected: screenshot-1.png is created in the project top level
   # Actual:   Build exception: 
   
   #    ~/tmp/screenshot/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java:19: error: package dev.flutter.plugins.integration_test does not exist
   #    flutterEngine.getPlugins().add(new dev.flutter.plugins.integration_test.IntegrationTestPlugin()); ^
   #    1 error
   #    FAILURE: Build failed with an exception. 
  
   # Note: renaming `example` to `example1` or any other name makes the test to work again.
```

## Supplemental information: The project https://github.com/mzimmerm/screenshot on Github was created as follows:

```shell
   cd ~/tmp
   flutter create --template=app screenshot
   # - Edited pubspec.yaml and added test, integration_test and flutter_test
   # - Added directory and file `integration_test/screenshot_test.dart`
   #   as described in https://github.com/flutter/flutter/tree/master/packages/integration_test#integration_test.
   #   (alternative in https://dev.to/mjablecnik/take-screenshot-during-flutter-integration-tests-435k is almost the same 
   #   and also does not work)
   # - Added directory and file `test_driver/integration_test.dart`
   #   as described in https://github.com/flutter/flutter/tree/master/packages/integration_test#integration_test.
   #   (alternative in https://dev.to/mjablecnik/take-screenshot-during-flutter-integration-tests-435k is almost the same 
   #   and also does not work)
   # - Added directory named `example1` for now
   # - Added project `screenshot` on Github
   #     `https://github.com/mzimmerm/screenshot` 
   # 
```
 
## Supplemental information: The result of the first successful test
 
```
     Running "flutter pub get" in screenshot...                         819ms
     Running Gradle task 'assembleDebug'...                             24.1s
     ✓  Built build/app/outputs/flutter-apk/app-debug.apk.
     Installing build/app/outputs/flutter-apk/app.apk...                618ms
     D/EGL_emulation( 4947): eglMakeCurrent: 0xe2a38b00: ver 3 0 (tinfo 0xda90fc90)
     VMServiceFlutterDriver: Connecting to Flutter application at http://127.0.0.1:34567/i7VCyPBOS58=/
     VMServiceFlutterDriver: Isolate found with number: 4413108995828315
     VMServiceFlutterDriver: Isolate is paused at start.
     VMServiceFlutterDriver: Attempting to resume isolate
     I/flutter ( 4947): 00:00 +0: screenshot
     VMServiceFlutterDriver: Connected to Flutter application.
     W/Gralloc3( 4947): allocator 3.x is not supported
     D/EGL_emulation( 4947): eglMakeCurrent: 0xda91a120: ver 3 0 (tinfo 0xda90fc10)
     D/EGL_emulation( 4947): eglMakeCurrent: 0xe2a38b00: ver 3 0 (tinfo 0xda90fc90)
     D/EGL_emulation( 4947): eglMakeCurrent: 0xe2a38b00: ver 3 0 (tinfo 0xda90fc90)
     D/EGL_emulation( 4947): eglCreateContext: 0xda91a7e0: maj 3 min 0 rcv 3
     D/EGL_emulation( 4947): eglMakeCurrent: 0xda91a7e0: ver 3 0 (tinfo 0xda90fc90)
     D/EGL_emulation( 4947): eglMakeCurrent: 0xe2a38b00: ver 3 0 (tinfo 0xda90fc90)
     D/EGL_emulation( 4947): eglMakeCurrent: 0xe2a38b00: ver 3 0 (tinfo 0xda90fc90)
     I/flutter ( 4947): 00:01 +1: (tearDownAll)
     W/mple.screensho( 4947): Accessing hidden method
     Lsun/misc/Unsafe;->compareAndSwapObject(Ljava/lang/Object;JLjava/lang/Object;Ljava/lang/Object;)Z
     (greylist, linking, allowed)
     I/flutter ( 4947): 00:01 +2: All tests passed!
     All tests passed.
```    
## Supplemental information: The result of the failed test when directory `example` exists:
 
```
    Running "flutter pub get" in screenshot...                       1,864ms
     ~/tmp/screenshot/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java:19: error: package dev.flutter.plugins.integration_test does not exist
           flutterEngine.getPlugins().add(new dev.flutter.plugins.integration_test.IntegrationTestPlugin());
                                                                                  ^
     1 error
     
     FAILURE: Build failed with an exception.
     
     * What went wrong:
     Execution failed for task ':app:compileDebugJavaWithJavac'.
     > Compilation failed; see the compiler error output for details.
     
     * Try:
     Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
     
     * Get more help at https://help.gradle.org
     
     BUILD FAILED in 10s
     Running Gradle task 'assembleDebug'...                             10.5s
     Gradle task assembleDebug failed with exit code 1
```

@mamuseferha This is still failing on master (in fact it is failing on all channels). After you switched from stable where you duplicated the issue, to master, did you upgrade and clean? I am including the exact steps I did and the log with the failure on master (same failure as on stable), can you please try them in this order, thanks:


Steps to reproduce on master: (this assumes you still have the screenshot project checked out)

```
- [X] cd screenshot. **Make sure directory 'example' exists**. Renaming 'example' to 'example1' or any other name, the issue disappears
- [X] flutter channel master
  - [X] Exit emulator
  - [X] flutter upgrade
  - [X] flutter emulator --launch "Nexus_6_API_29_2"; sleep 20 # YOUR EMULATOR WILL DIFFER
  - [X] flutter clean; flutter pub upgrade; flutter pub get; flutter drive --driver=test_driver/integration_test.dart --target=integration_test/screenshot_test.dart
  - [X] *failed*
```
Failure logs

```
screenshot> flutter clean; flutter pub upgrade; flutter pub get; flutter drive --driver=test_driver/integration_test.dart --target=integration_test/screenshot_test.dart
Deleting build...                                                   21ms
Deleting .dart_tool...                                               2ms
Deleting .packages...                                                0ms
Resolving dependencies...
  _fe_analyzer_shared 31.0.0 (32.0.0 available)
  analyzer 2.8.0 (3.0.0 available)
  archive 3.1.6
  args 2.3.0
  async 2.8.2
  boolean_selector 2.1.0
  characters 1.2.0
  charcode 1.3.1
  cli_util 0.3.5
  clock 1.1.0
  collection 1.15.0
  convert 3.0.1
  coverage 1.0.3
  crypto 3.0.1
  cupertino_icons 1.0.4
  fake_async 1.2.0
  file 6.1.2
  flutter 0.0.0 from sdk flutter
  flutter_driver 0.0.0 from sdk flutter
  flutter_lints 1.0.4
  flutter_test 0.0.0 from sdk flutter
  frontend_server_client 2.1.2
  fuchsia_remote_debug_protocol 0.0.0 from sdk flutter
  glob 2.0.2
  http_multi_server 3.0.1
  http_parser 4.0.0
  integration_test 0.0.0 from sdk flutter
  io 1.0.3
  js 0.6.3
  lints 1.0.1
  logging 1.0.2
  matcher 0.12.11
> material_color_utilities 0.1.3 (was 0.1.2)
  meta 1.7.0
  mime 1.0.1
  node_preamble 2.0.1
  package_config 2.0.2
  path 1.8.0 (1.8.1 available)
  platform 3.1.0
  pool 1.5.0
  process 4.2.4
  pub_semver 2.1.0
  shelf 1.2.0
  shelf_packages_handler 3.0.0
  shelf_static 1.1.0
  shelf_web_socket 1.0.1
  sky_engine 0.0.99 from sdk flutter
  source_map_stack_trace 2.1.0
  source_maps 0.10.10
  source_span 1.8.1
  stack_trace 1.10.0
  stream_channel 2.1.0
  string_scanner 1.1.0
  sync_http 0.3.0
  term_glyph 1.2.0
  test 1.19.5 (1.20.0 available)
  test_api 0.4.8 (0.4.9 available)
  test_core 0.4.9 (0.4.10 available)
  typed_data 1.3.0
  vector_math 2.1.1
  vm_service 7.5.0 (8.1.0 available)
  watcher 1.0.1
  web_socket_channel 2.1.0
  webdriver 3.0.0
  webkit_inspection_protocol 1.0.0
  yaml 3.1.0
Downloading material_color_utilities 0.1.3...
Changed 1 dependency!
7 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
Running "flutter pub get" in screenshot...                         791ms
Running "flutter pub get" in screenshot...                         783ms
/home/mzimmermann/dev/my-projects-source/public-on-github/screenshot/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java:19: error: package dev.flutter.plugins.integration_test does not exist
      flutterEngine.getPlugins().add(new dev.flutter.plugins.integration_test.IntegrationTestPlugin());
                                                                             ^
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 16s
Running Gradle task 'assembleDebug'...                             16.8s
Gradle task assembleDebug failed with exit code 1
```

@mzimmerm, running the command all at once seemed to do the trick. Issue is reproducible on master and stable channel

<details>
<summary>logs</summary>

```bash

flutter clean; flutter pub upgrade; flutter pub get; flutter drive --driver=test_driver/integration_test.dart --target=integration_test/screenshot_test.dart
Cleaning Xcode workspace...                                      2,889ms
Deleting build...                                                  125ms
Deleting .dart_tool...                                               5ms
Deleting .packages...                                                1ms
Deleting Generated.xcconfig...                                       0ms
Deleting flutter_export_environment.sh...                            0ms
Deleting .flutter-plugins-dependencies...                            0ms
Deleting .flutter-plugins...                                         0ms
Resolving dependencies...
  _fe_analyzer_shared 31.0.0 (32.0.0 available)
  analyzer 2.8.0 (3.0.0 available)
  archive 3.1.6
  args 2.3.0
  async 2.8.2
  boolean_selector 2.1.0
  characters 1.2.0
  charcode 1.3.1
  cli_util 0.3.5
  clock 1.1.0
  collection 1.15.0
  convert 3.0.1
  coverage 1.0.3 (1.0.4 available)
  crypto 3.0.1
  cupertino_icons 1.0.4
  fake_async 1.2.0
  file 6.1.2
  flutter 0.0.0 from sdk flutter
  flutter_driver 0.0.0 from sdk flutter
  flutter_lints 1.0.4
  flutter_test 0.0.0 from sdk flutter
  frontend_server_client 2.1.2
  fuchsia_remote_debug_protocol 0.0.0 from sdk flutter
  glob 2.0.2
  http_multi_server 3.0.1
  http_parser 4.0.0
  integration_test 0.0.0 from sdk flutter
  io 1.0.3
  js 0.6.3
  lints 1.0.1
  logging 1.0.2
  matcher 0.12.11
  material_color_utilities 0.1.3
  meta 1.7.0
  mime 1.0.1
  node_preamble 2.0.1
  package_config 2.0.2
  path 1.8.0 (1.8.1 available)
  platform 3.1.0
  pool 1.5.0
  process 4.2.4
  pub_semver 2.1.0
  shelf 1.2.0
  shelf_packages_handler 3.0.0
  shelf_static 1.1.0
  shelf_web_socket 1.0.1
  sky_engine 0.0.99 from sdk flutter
  source_map_stack_trace 2.1.0
  source_maps 0.10.10
  source_span 1.8.1
  stack_trace 1.10.0
  stream_channel 2.1.0
  string_scanner 1.1.0
  sync_http 0.3.0
  term_glyph 1.2.0
  test 1.19.5 (1.20.1 available)
  test_api 0.4.8 (0.4.9 available)
  test_core 0.4.9 (0.4.11 available)
  typed_data 1.3.0
  vector_math 2.1.1
  vm_service 7.5.0 (8.1.0 available)
  watcher 1.0.1
  web_socket_channel 2.1.0
  webdriver 3.0.0
  webkit_inspection_protocol 1.0.0
  yaml 3.1.0
No dependencies changed.
8 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
Running "flutter pub get" in screenshot...                       1,616ms
Running "flutter pub get" in screenshot...                       1,962ms
Multiple devices found:
sdk gphone x86 (mobile) • emulator-5554                        • android-x86    • Android 11 (API 30) (emulator)
iPhone XR (mobile)      • 52310930-1602-4A61-BE82-9B18989453E0 • ios            •
com.apple.CoreSimulator.SimRuntime.iOS-15-0 (simulator)
macOS (desktop)         • macos                                • darwin-x64     • macOS 12.1 21C52 darwin-x64
Chrome (web)            • chrome                               • web-javascript • Google Chrome 93.0.4577.63
[1]: sdk gphone x86 (emulator-5554)
[2]: iPhone XR (52310930-1602-4A61-BE82-9B18989453E0)
[3]: macOS (macos)
[4]: Chrome (chrome)
Please choose one (To quit, press "q/Q"): 1
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
/Users/flo/projects/oos/screenshot/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java:19: error: package dev.flutter.plugins.integration_test does not exist
      flutterEngine.getPlugins().add(new dev.flutter.plugins.integration_test.IntegrationTestPlugin());
                                                                             ^
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 14s
Running Gradle task 'assembleDebug'...                             15.4s
Gradle task assembleDebug failed with exit code 1
```

</details>

<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel stable, 2.8.1, on macOS 12.1 21C52 darwin-x64, locale en-US)
    • Flutter version 2.8.1 at /Users/flo/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 77d935af4d (4 days ago), 2021-12-16 08:37:33 -0800
    • Engine revision 890a5fca2e
    • Dart version 2.15.1

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/flo/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.62.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (4 available)
    • sdk gphone x86 (mobile) • emulator-5554                        • android-x86    • Android 11 (API 30) (emulator)
    • iPhone 13 (mobile)      • EF1251D3-C9C4-475E-9F91-58CD010FDA48 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-15-0 (simulator)
    • macOS (desktop)         • macos                                • darwin-x64     • macOS 12.1 21C52 darwin-x64
    • Chrome (web)            • chrome                               • web-javascript • Google Chrome 93.0.4577.63

• No issues found!
```
```bash
[✓] Flutter (Channel master, 2.9.0-1.0.pre.144, on macOS 12.1 21C52 darwin-x64, locale en-US)
    • Flutter version 2.9.0-1.0.pre.144 at /Users/flo/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 07a36f07e3 (2 hours ago), 2021-12-21 01:09:07 -0500
    • Engine revision d3075cd085
    • Dart version 2.16.0 (build 2.16.0-123.0.dev)
    • DevTools version 2.9.1

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/flo/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.62.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (4 available)
    • sdk gphone x86 (mobile) • emulator-5554                        • android-x86    • Android 11 (API 30)
      (emulator)
    • iPhone XR (mobile)      • 52310930-1602-4A61-BE82-9B18989453E0 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-15-0 (simulator)
    • macOS (desktop)         • macos                                • darwin-x64     • macOS 12.1 21C52 darwin-x64
    • Chrome (web)            • chrome                               • web-javascript • Google Chrome 93.0.4577.63

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

Thank you

--- a/packages/flutter_tools/lib/src/project.dart
+++ b/packages/flutter_tools/lib/src/project.dart
@@ -310,7 +310,7 @@ class FlutterProject {
     bool winUwpPlatform = false,
     DeprecationBehavior deprecationBehavior = DeprecationBehavior.none,
   }) async {
-    if (!directory.existsSync() || hasExampleApp || isPlugin) {
+    if (!directory.existsSync() || isPlugin) {
       return;
     }
     await refreshPluginsList(this, iosPlatform: iosPlatform, macOSPlatform: macOSPlatform);

--- a/packages/flutter_tools/test/general.shard/project_test.dart
+++ b/packages/flutter_tools/test/general.shard/project_test.dart
@@ -156,6 +156,18 @@ void main() {
         expectNotExists(project.ios.hostAppRoot.childDirectory('Flutter').childFile('Generated.xcconfig'));
         expectNotExists(project.android.hostAppGradleRoot.childFile('local.properties'));
       });
+      _testInMemory('works if there is an "example" folder', () async {
+        final FlutterProject project = await someProject();
+        // The presence of an "example" folder used to be used as an indicator
+        // that a project was a plugin, but shouldn't be as this creates false
+        // positives.
+        project.directory.childDirectory('example').createSync();
+        await project.regeneratePlatformSpecificTooling();
+        expectExists(project.ios.hostAppRoot.childDirectory('Runner').childFile('GeneratedPluginRegistrant.h'));
+        expectExists(androidPluginRegistrant(project.android.hostAppGradleRoot.childDirectory('app')));
+        expectExists(project.ios.hostAppRoot.childDirectory('Flutter').childFile('Generated.xcconfig'));
+        expectExists(project.android.hostAppGradleRoot.childFile('local.properties'));
+      });
       _testInMemory('injects plugins for iOS', () async {
         final FlutterProject project = await someProject();
         await project.regeneratePlatformSpecificTooling();

